api/v1alpha3/azureimage_validation.go

```diff
@@ -44,6 +44,18 @@ func ValidateImage(image *Image, fldPath *field.Path) field.ErrorList {
 	return allErrs
 }
 
+// ValidateSSHKey validates an SSHKey
+func ValidateSSHKey(sshKey string, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if sshKey == "" {
```

Comment:
I don't think this should validate that the key is not empty, since we allow empty keys (and default them in the defaulting webhook, so the key should never be empty here). Instead, we should check that if the key is user-provided (ie. not empty) it is a valid SSH key. I think you might be able to use ssh.ParsePublicKey() to do that https://github.com/golang/crypto/blob/master/ssh/keys.go#L265:6